HousingFinanceInterimApi.Tests/V1/UseCase/LoadActionDiaryUseCaseTests.cs

```diff
@@ -0,0 +1,166 @@
+
+    public LoadActionDiaryUseCaseTests()
+    {
+        _mockBatchLogGateway = new Mock<IBatchLogGateway>();
+        _mockBatchLogErrorGateway = new Mock<IBatchLogErrorGateway>();
+        _mockActionDiaryGateway = new Mock<IActionDiaryGateway>();
+        _mockGoogleFileSettingGateway = new Mock<IGoogleFileSettingGateway>();
+        _mockGoogleClientService = new Mock<IGoogleClientService>();
+
+        var waitDurationEnvVar = 30;
+        Environment.SetEnvironmentVariable("WAIT_DURATION", waitDurationEnvVar.ToString());
+
+        _classUnderTest = new LoadActionDiaryUseCase
+            (
+                _mockBatchLogGateway.Object,
+                _mockBatchLogErrorGateway.Object,
+                _mockActionDiaryGateway.Object,
+                _mockGoogleFileSettingGateway.Object,
+                _mockGoogleClientService.Object
+            );
+
+        _mockBatchLogGateway
+            .Setup(g => g.CreateAsync(It.IsAny<string>(), It.IsAny<bool>()))
+            .ReturnsAsync(RandomGen.BatchLogDomain());
+
+        _mockBatchLogGateway
+            .Setup(g => g.CreateAsync(It.IsAny<string>(), It.IsAny<bool>()))
+            .ReturnsAsync(RandomGen.BatchLogDomain());
+
+        var rentPositionFileSettings = RandomGen.CreateMany<GoogleFileSettingDomain>(quantity: 1).ToList();
+        _mockGoogleFileSettingGateway
+            .Setup(g => g.GetSettingsByLabel(It.IsAny<string>()))
+            .ReturnsAsync(rentPositionFileSettings);
+
+        _mockGoogleClientService
+            .Setup(x => x.GetFilesInDriveAsync(It.IsAny<string>(), It.IsAny<string>()))
+            .ReturnsAsync(RandomGen.CreateMany<File>(1).ToList());
+
+        _mockGoogleClientService
+            .Setup(x => x.ReadSheetToEntitiesAsync<ActionDiaryAuxDomain>(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<string>())
+            ).ReturnsAsync(RandomGen.CreateMany<ActionDiaryAuxDomain>(quantity: 1).ToList());
+    }
+
+    [Fact]
+    public async Task ReturnsStepResponseWhenAllOK()
+    {
+        // Act
+        var useCaseCall = async () => await _classUnderTest.ExecuteAsync().ConfigureAwait(false);
+
+        // Assert
+        await useCaseCall.Should().NotThrowAsync().ConfigureAwait(false);
+
+        var stepResponse = await useCaseCall().ConfigureAwait(false);
+        stepResponse.Continue.Should().BeTrue();
+    }
+
+    [Fact]
+    public async Task LoadsActionDiaryWhenAllOK()
+    {
+        // Act
+        var useCaseCall = async () => await _classUnderTest.ExecuteAsync().ConfigureAwait(false);
+
+        // Assert
+        var result = await useCaseCall().ConfigureAwait(false);
+        result.Continue.Should().BeTrue();
+
+        _mockActionDiaryGateway.Verify(gw =>
+                gw.ClearActionDiaryAuxiliary(), Times.Once);
+        _mockActionDiaryGateway.Verify(gw =>
+                gw.CreateBulkAsync(It.IsAny<IList<ActionDiaryAuxDomain>>()),
```

Comment:
You make this assertion more extensive by testing that the `CreateBulk` method gets called with the expected `ActionDiaryAuxDomain`.

So, in case someone messes up while refactoring in the future, should the `CreateBulk` method receive a wrong list or a `null` list, your test would be able to detect that.